### PR TITLE
Create FlashProgram from buffer data.

### PR DESCRIPTION
**Pull request by arekzaluski**
LineBreakRule was changed to enable it on Windows. Can we have a different rules per Windows and Unix ?

### CODE REVIEW

File: src/targets/FlashProgram.ts [hunk 1]
```diff
@@ -38,6 +38,19 @@ export class FlashProgram {
 
     constructor(public sections: FlashSection[]) {}
 
+    public static fromArrayBuffer(buffer: ArrayBuffer): FlashProgram {
+        // detect if buffer contains text or binary data
+        const lengthToCheck = buffer.byteLength > 50 ? 50 : buffer.byteLength;
+        const bufferString = Buffer.from(buffer).toString("utf8");
+        for (let i = 0; i < lengthToCheck; i++) {
+            const charCode = bufferString.charCodeAt(i);
+            if (charCode === 65533 || charCode <= 8) {
```
Comment (arekzaluski) [inline src/targets/FlashProgram.ts]: 65533 means unknown character and under 8 are control characters. I can add it as a comment above if you want. Tested on k64f and Microbit.

**Comment by arekzaluski** [comment]
Added it only for Node.js example. De we want also to use it in web example?

File: src/targets/FlashProgram.ts [hunk 2]
```diff
@@ -38,6 +38,21 @@ export class FlashProgram {
 
     constructor(public sections: FlashSection[]) {}
 
+    public static fromArrayBuffer(buffer: ArrayBuffer): FlashProgram {
+        // detect if buffer contains text or binary data
```
Comment (thegecko) [inline src/targets/FlashProgram.ts]: nit: this check should ideally be it's own method

**Comment by thegecko** [comment]
I think we should add it to the web example, too. Happy to merge this for now, though. Have tested both k64f and micro:bit from web and node and all works fine.

File: src/targets/FlashProgram.ts (hunk 2 above)
Comment (arekzaluski) [reply src/targets/FlashProgram.ts]: I'll add this change in other PR.

**Pull request merged**